examples/vision/mirnet.py

```diff
@@ -0,0 +1,508 @@
+
+
+model = mirnet_model(num_rrg=3, num_mrb=2, channels=64)
+
+"""
+## Training
+
+- We train MIRNet using **Charbonnier Loss** as the loss function and **Adam
+Optimizer** with a learning rate of `1e-4`.
+- We use **Peak Signal Noise Ratio** or PSNR as a metric which is an expression for the
+ratio between the maximum possible value (power) of a signal and the power of distorting
+noise that affects the quality of its representation.
+"""
+
+
+def charbonnier_loss(y_true, y_pred):
+    return tf.reduce_mean(tf.sqrt(tf.square(y_true - y_pred) + tf.square(1e-3)))
+
+
+def peak_signal_noise_ratio(y_true, y_pred):
+    return tf.image.psnr(y_pred, y_true, max_val=255.0)
+
+
+optimizer = keras.optimizers.Adam(learning_rate=1e-4)
+model.compile(
+    optimizer=optimizer, loss=charbonnier_loss, metrics=[peak_signal_noise_ratio]
+)
+
+history = model.fit(
+    train_dataset,
+    validation_data=val_dataset,
+    epochs=50,
+    callbacks=[
+        keras.callbacks.ReduceLROnPlateau(
+            monitor="val_peak_signal_noise_ratio",
+            factor=0.5,
+            patience=5,
+            verbose=1,
+            min_delta=1e-7,
+            mode="max",
+        )
+    ],
+)
+
+plt.plot(history.history["loss"], label="train_loss")
+plt.plot(history.history["val_loss"], label="val_loss")
+plt.xlabel("Epochs")
+plt.ylabel("Loss")
+plt.title("Train and Validation Losses Over Epochs", fontsize=14)
+plt.legend()
+plt.grid()
+plt.show()
+
+
+plt.plot(history.history["peak_signal_noise_ratio"], label="train_psnr")
+plt.plot(history.history["val_peak_signal_noise_ratio"], label="val_psnr")
+plt.xlabel("Epochs")
+plt.ylabel("PSNR")
+plt.title("Train and Validation PSNR Over Epochs", fontsize=14)
+plt.legend()
+plt.grid()
+plt.show()
+
+"""
+## Inference
+"""
+
+
+def plot_results(images, titles, figure_size=(12, 12)):
+    fig = plt.figure(figsize=figure_size)
+    for i in range(len(images)):
+        fig.add_subplot(1, len(images), i + 1).set_title(titles[i])
+        _ = plt.imshow(images[i])
+        plt.axis("off")
+    plt.show()
+
+
+def infer(original_image):
+    image = keras.preprocessing.image.img_to_array(original_image)
+    image = image.astype("float32") / 255.0
+    image = np.expand_dims(image, axis=0)
+    output = model.predict(image)
+    output_image = output[0] * 255.0
+    output_image = output_image.clip(0, 255)
+    output_image = output_image.reshape(
+        (np.shape(output_image)[0], np.shape(output_image)[1], 3)
+    )
+    output_image = Image.fromarray(np.uint8(output_image))
+    original_image = Image.fromarray(np.uint8(original_image))
+    return output_image
+
+
+"""
+### Inference on Train Images
+"""
+
+for low_light_image in random.sample(train_low_light_images, 4):
+    original_image = Image.open(low_light_image)
+    enhanced_image = infer(original_image)
+    plot_results([original_image, enhanced_image], ["Original", "Enhanced"])
+
+"""
+### Inference on Validation Images
```

Comment:
You can also remove this section since you have a test section below

Reply:
Removed inference on validation images.